pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -136,6 +137,10 @@ public PulsarRecordCursor(List<PulsarColumnHandle> columnHandles, PulsarSplit pu
                         pulsarConnectorConfig),
                 new PulsarConnectorMetricsTracker(pulsarConnectorCache.getStatsProvider()));
         this.decoderFactory = decoderFactory;
+        if (pulsarConnectorConfig.getMaxSplitEntryQueueSizeBytes() >= 0) {
+            this.entryCacheSizeAllocator = new CacheSizeAllocator(
```

Comment:
I think you can make it an interface. Then have a no-op implementation and a CacheSizeAllocator implementation. Then you don't need to add `if (entryCacheSizeAllocator != null)` everywhere else.

Reply:
Yes, that seems better.